packages/falter-berlin-ssid-changer/files/lib/lib_ssid-changer.sh

```diff
@@ -1,22 +1,64 @@
 #/bin/sh
 . /lib/functions.sh
 
+DNS_SERVER="8.8.8.8 1.1.1.1 9.9.9.9"
 NODENAME=$(uci_get system @system[-1] hostname | cut -b -24 ) # cut nodename to not exceed 32 bytes
 OFFLINE_SSID="offline_""$NODENAME"
 
-is_internet_reachable() {
-    # check if we really lost internet connection.
-    local ERROR=0
-    local DNS_SERVER="8.8.8.8 1.1.1.1 9.9.9.9"
+
+log() {
+    logger -s -t "ssid-changer" -p 5 "$1"
+}
+
+increment_ip_addr() {
+    # ToDo: Implement actual function
+    local raw_addr="$1"
+
+    local net=$(echo $raw_addr | cut -d'.' -f -3)
+    local host=$(echo $raw_addr | cut -d'.' -f 4)
+    host=$(( $host + 1 ))
+    echo "$net"".$host"
+}
+
+is_internet_reachable_via_ffuplink() {
+    # check if clients could reach the internet via ffuplink (only exists if we have a wan port)
+    for SERVER in $DNS_SERVER; do
+        ping -c1 -W 3 -I ffuplink $SERVER > /dev/null
+        if [ $? = 0 ]; then
+            return 0
+        fi
+    done
+    return 1
+}
+
+# ip r g 9.9.9.9 from 172.16.2.2 iif br-dhcp
+# DHCP_CLIENT_ADDR=$(uci_get dhcp frei_funk ip)
+# ip r g $SRV from 172.16.2.2 iif br-dhcp
+
+exists_route_over_mesh() {
+    # if internet not reachable via ffuplink, is there a router over the mesh?
+    local dhcp=$(uci_get dhcp frei_funk ip)
```

Comment:
in /lib/functions/network.sh there is already a call to get the ip address of a network interface.  You could call network_get_ipaddr("dhcp") and reuse their code.  In fact, you could end up with
```local dhcp=$(increment_ip_addr(network_get_ipaddr("dhcp")))```
which I think is very readable.

Reply:
Thats right. But as it would save only one line, I'd rather prefer to not include a further dependency.

Reply:
It depends on what you think of as a dependency.  The function ```network_get_ipaddr``` is a part of openwrt's core system.  But I will leave the decision up to you.

Reply:
I changed it to your approach in the new version: looks a bit more nice, maybe.

Reply:
I think you might still need to push the changes.  your last push was 10 days ago